src/main/scala/com.snowplowanalytics.maxmind.iplookups/IpLookups.scala

```diff
@@ -51,6 +52,11 @@ object IpLookups {
       memCache,
       lruCache
     )
+
+  implicit class RichValidated[E, A](validated: Validated[E, A]) {
+    def flatMap[EE >: E, B](f: A => Validated[EE, B]): Validated[EE, B] =
+      validated.andThen(f)
+  }
```

Comment:
I invite you to read https://typelevel.org/cats/datatypes/validated.html#of-flatmaps-and-eithers on why `flatMap` on `Validated` is not a good idea.

The problem is that it was defined on Scalaz' `Validation` and we've been abusing it a lot throughout the the different codebases.

Here we don't actually need the error-accumulating behaviour of `Validated` so I think we can safely move to `Either`.

Reply:
Ok, checked! I'm going to read now! Thank you Ben